### PR TITLE
Switch to core22 base

### PR DESCRIPTION
**Pull request by jocado**
Make required changes to enable snap build with core22 base

Relates to: https://github.com/docker-snap/docker-snap/issues/65

### CODE REVIEW

**Review by xnox — APPROVED** [review]
But let's see how our ci will react to things.

**Comment by jocado** [comment]
I've pushed a change to try and fix the previous workflow failure, but it seems you have to approve each run @xnox

**Comment by jocado** [comment]
I guess its because the branch is in my own fork

**Comment by jocado** [comment]
Snap was built at least.

I'm not 100% how to interpret the failure: https://github.com/docker-snap/docker-snap/actions/runs/4409275831/jobs/7725502411#step:10:6

Do you have any thoughts @xnox ?

Worth noting that the snap I built locally ran fine on a UC22 host, tested with hello-world docker image.

I suppose if this was my own CI permission in, I would add some debug to the workflow and run it again.

**Comment by tianon** [comment]
A core22 snap is not supported on an Ubuntu 20.04 system, right?  If it is, then this failure is worth digging more into, but if it isn't then you probably just want to update this line to `ubuntu-22.04` instead:

https://github.com/docker-snap/docker-snap/blob/975f02462d78b74b96da4629dff480d14ec57542/.github/workflows/smoke-test.yml#L14

**Comment by xnox** [comment]
> A core22 snap is not supported on an Ubuntu 20.04 system, right?  If it is, then this failure is worth digging more into, but if it isn't then you probably just want to update this line to `ubuntu-22.04` instead:
> 
> https://github.com/docker-snap/docker-snap/blob/975f02462d78b74b96da4629dff480d14ec57542/.github/workflows/smoke-test.yml#L14

We could update to 22.04 host base for workflows. But in theory it should work. We have Lxd snap built on any base, and running on any host. Including when Lxd snap was base:core20 and when it moved to base:core22.

**Comment by tianon** [comment]
Ah, so the failure probably _is_ worth investigating then, because that's even the most basic use case of `docker run hello-world` failing :sweat_smile:

**Comment by xnox** [comment]
> Ah, so the failure probably _is_ worth investigating then, because that's even the most basic use case of `docker run hello-world` failing sweat_smile

i think it might be fun things between focal host, core22 confinement, and changes in apparmor_parser --version which tries to parse things, and gets a denial to access /abi/ subdir, that is there, but not allowed to by the docker snapd interface.

This probably works on UC because we don't ship /abi/ directory on UC and nothing tries to access it either. Fun things.

Somewhat wonder if we should try upgrade to core20 first, cause fixing this issue may require upgrades to docker itself, and/or changes to the snapd interfaces.

**Review by xnox — CHANGES_REQUESTED** [review]
Block merging until we get feedback from interface maintainers and apparmor maintainers.

**Comment by jocado** [comment]
@xnox Just to join the dots - what was the feedback from interface maintainers and apparmor maintainers ?

**Pull request merged**
